### DISCUSSION TITLE
MGCB Editor not opening on M3 Mac with Sonoma

### DISCUSSION THREAD

**Discussion by sim8** [MonoGame/MonoGame#8469] [Content]
Hi all, I'm not able to open the MGCB editor at all and wondering if anyone else has steps I can try. So far I've tried:

- Installing x64 .NET SDK (`dotnet --version`: `8.0.401`)
- Adding `export PATH="/usr/local/share/dotnet/x64:$PATH"` to `zshrc`
- Running `dotnet mgcb-editor ./Content/Content.mgcb` from the Project level - it doesn't open, and no errors show in logs

Any suggestions appreciated!

**Comment by stromkos** [comment]
M3 is not x64. It is Arm64.  

Uninstall the X64 version. If all else fails: `rm -rf /usr/local/share/dotnet/x64` from the terminal window.
Please remove the PATH from the `zshrc` file. Update after install, if needed.


https://dotnet.microsoft.com/en-us/download/dotnet/8.0

Or direct link to the .Net 8 installer Mac Arm64 download: https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-8.0.401-macos-arm64-installer

Follow the Mac install instructions from the website again.

I believe the Mac Arm64 code base is complete. YMMV.

**Comment by sim8** [comment]
Thanks! I should've mentioned that I initially installed the Arm64 SDK + without the PATH entry and was having the same issue - I was able to build Monogame but the MGCB editor did not open. That's what led me to following these instructions which didn't work for me: https://community.monogame.net/t/tutorial-for-setting-up-monogame-on-m1-m2-apple-silicon/19669

**Comment by sim8** [comment]
Just to confirm I've tried this again with both x64 and Arm64 installations of .NET 8. In both cases I see the application icon for the MGCB editor flash up in the dock then disappear, without any console errors.

Is it still recommended for Apple Silicon users to stick with x64? https://docs.monogame.net/articles/getting_started/1_setting_up_your_os_for_development_macos.html?tabs=android#:~:text=Apple%20Silicon%20Known%20Issues

**Reply by dellis1972** [reply]
The fixes for the arm processors will be released in 3.8.3 , we have only just merged the commit which will let this work.

**Reply by sim8** [reply]
Amazing, thanks!

**Comment by jazzay** [comment]
@dellis1972 I'm having this same issue. Do you know when 3.8.3 will be released?

**Comment by dellis1972** [comment]
@jazzay I think the plan is in about 3 weeks. 
If you would like to try the development packages I have written some documentation (its not merged yet) at https://github.com/MonoGame/docs.monogame.github.io/blob/84d558f37e7d5324561a4d0c9d99e7a0c7ff4d2f/articles/getting_started/using_development_nuget_packages.md. 

This would allow you to test the latest. Note there may well be some issues still as we are still working on some bugs, but it might help. 

Interestingly I get the same issue on 3.8.2 , so I will investigate that.

**Reply by jazzay** [reply]
Thanks for the info on development packages. Will give that a try.

**Reply by dellis1972** [reply]
I did some investingation and it won't work. We are not even shipping an arm64 version of the editor. Its only x64.
Add to that there is a bug in Eto.Forms which stops the x64 app running on an arm64 machine even if you run it in a x64 dotnet. 
I am working on a fix now that I know what the issue is.

**Comment by dellis1972** [comment]
I just opened a PR which should allow for an arm64 build of the editor https://github.com/MonoGame/MonoGame/pull/8505.

**Reply by jazzay** [reply]
Awesome!

**Comment by jazzay** [comment]
@dellis1972 I see that your Mac editor pr #8505 was merged, and it has been about 3 weeks. Do you all have a sense of when the next release (3.8.3) will land?

**Comment by ovoelke** [comment]
Is this problem still being addressed at all?

I use a MacBook Pro with M1 Pro. For me, MonoGame is not usable as MGCB is an essential part of the toolchain. MGCB-EDIT doesn't work for me either.
